rust/azure_iot_operations_connector/src/base_connector.rs

```diff
@@ -6,17 +6,53 @@
 use std::{sync::Arc, time::Duration};
 
 use azure_iot_operations_mqtt::session::{
-    Session, SessionError, SessionManagedClient, SessionOptionsBuilder,
+    AIOBrokerFeatures, AIOBrokerFeaturesBuilder, Session, SessionError, SessionManagedClient,
+    SessionOptionsBuilder,
+    reconnect_policy::{ExponentialBackoffWithJitter, ReconnectPolicy},
 };
 use azure_iot_operations_protocol::application::ApplicationContext;
 use azure_iot_operations_services::{azure_device_registry, schema_registry, state_store};
+use derive_builder::Builder;
 use managed_azure_device_registry::DeviceEndpointClientCreationObservation;
 
 use crate::deployment_artifacts::connector::ConnectorArtifacts;
 
 pub mod adr_discovery;
 pub mod managed_azure_device_registry;
 
+/// Options for configuring a new [`BaseConnector`]
+#[derive(Builder)]
+#[builder(pattern = "owned")]
+pub struct Options {
+    // Options for configuring the MQTT session
+    /// Maximum number of outgoing messages not yet accepted by the
+    #[builder(default = "100")]
+    outgoing_max: usize,
```

Comment:
is there any way we can do these so that the default is linked to the default in the SessionOptions rather than duplicating them here? I think the only way to do that though would be to have these be
```
#[builder(default = "None")]
outgoing_max: Option<usize>,
```
and then if it's None, don't pass it to the SessionOptions builder later on, but that kinda feels more gross and harder for the customer to know what the default values will be?